calculator/src/CalculatorDigits.jsx

```diff
@@ -0,0 +1,54 @@
+/** @jsxImportSource @emotion/react */
+// import { useState } from "react";
+import { css } from "@emotion/react";
+
+const MAX_DIGITS_ON_DISPLAY = 6;
+
+function CalculatorDigits({ calculatorDisplay, setCalculatorDisplay }) {
+  return (
+    <div
+      css={css`
+        display: flex;
+        flex-direction: row;
+        flex-wrap: wrap;
+        justify-content: space-between;
+      `}
+    >
+      {[1, 2, 3, 4, 5, 6, 7, 8, 9, 0].map((digit) => (
+        <DigitButton
+          digit={digit}
+          setCalculatorDisplay={setCalculatorDisplay}
+          calculatorDisplay={calculatorDisplay}
+        />
+      ))}
+    </div>
+  );
+}
+
+function DigitButton({ digit, calculatorDisplay, setCalculatorDisplay }) {
+  function inputDigitToDisplay() {
+    if (calculatorDisplay.toString().length >= MAX_DIGITS_ON_DISPLAY) return;
```

Comment:
The button should not know what happens in the rest of the calculator to determine whether it should be pressable or not. What it can do - is receive a prop from above telling whether it is disabled or not. This is one of the most important things to grasp in React (or component-based UI for that matter, or technically in any modular system) - what business logic should the module (component in our case) be exposed to